ifrs17/Utils/EqualityComparers.ipynb

```diff
@@ -140,6 +140,35 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
+            "source": [
+                "class ParametersComparer: IEqualityComparer<DataNodeParameter>",
+                "\n{",
+                "\n    private ParametersComparer(){}",
+                "\n",
+                "\n    public bool Equals(DataNodeParameter x, DataNodeParameter y)",
+                "\n    {",
+                "\n        if (x == null && y == null) return true; ",
+                "\n        if (x == null || y == null) return false; ",
+                "\n        if (!(x.Year == y.Year && x.Month == y.Month && x.DataNode == y.DataNode && x.Scenario == y.Scenario)) return false; ",
+                "\n        if (x is SingleDataNodeParameter && y is SingleDataNodeParameter &&",
+                "\n            ((SingleDataNodeParameter)x).PremiumAllocation == ((SingleDataNodeParameter)y).PremiumAllocation) return true; ",
+                "\n        if (x is InterDataNodeParameter && y is InterDataNodeParameter &&",
+                "\n            ((InterDataNodeParameter)x).LinkedDataNode == ((InterDataNodeParameter)y).LinkedDataNode && ",
```

Comment:
DataNode and LinkedDataNode might be inverted but the comparer should return true